cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsTableMigration.java

```diff
@@ -82,67 +84,69 @@ public boolean isOldMetricsDataAvailable() {
    * @param maxRecordsToScan - limit on number of records scanned
    */
   public void transferData(int maxRecordsToScan) {
-    LOG.trace("Running Metrics Migrator Thread");
-    Scanner scanner = v2MetricsTable.scan(null, null, null);
-    Row row;
-    int recordsScanned = 0;
-
-    while (((row = scanner.next()) != null) && (recordsScanned < maxRecordsToScan)) {
-      if (recordsScanned % 10 == 0) {
-        LOG.trace("Scanned {} records in Metrics Data Migration", recordsScanned);
-      }
+    try (Scanner scanner = v2MetricsTable.scan(null, null, null)) {
+      LOG.debug("Starting scanning for Metrics Data Migration with {} max records", maxRecordsToScan);
+      Row row;
+      int recordsScanned = 0;
+      while ((recordsScanned < maxRecordsToScan) && ((row = scanner.next()) != null)) {
+        if (recordsScanned % 10 == 0) {
+          LOG.debug("Scanned {} records in Metrics Data Migration", recordsScanned);
+        }
 
-      byte[] rowKey = row.getRow();
-      Map<byte[], byte[]> columns = row.getColumns();
-      //row-map
-      SortedMap<byte[], SortedMap<byte[], Long>> rowGauges = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-      NavigableMap<byte[], NavigableMap<byte[], Long>> rowIncrements = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-
-      // column-map gauges
-      SortedMap<byte[], Long> gauges = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-      // column-map increments
-      NavigableMap<byte[], Long> increments = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-
-      for (Map.Entry<byte[], byte[]> entry : columns.entrySet()) {
-        // column is timestamp, do a get on the new table
-        byte[] value = v3MetricsTable.get(rowKey, entry.getKey());
-        if (value == null) {
-          gauges.put(entry.getKey(), Bytes.toLong(entry.getValue()));
-        } else {
-          increments.put(entry.getKey(), Bytes.toLong(entry.getValue()));
+        byte[] rowKey = row.getRow();
+        Map<byte[], byte[]> columns = row.getColumns();
+        //row-map
+        SortedMap<byte[], SortedMap<byte[], Long>> rowGauges = new TreeMap<>(Bytes.BYTES_COMPARATOR);
+        NavigableMap<byte[], NavigableMap<byte[], Long>> rowIncrements = new TreeMap<>(Bytes.BYTES_COMPARATOR);
+
+        // column-map gauges
+        List<byte[]> gaugeDeletes = new ArrayList<>();
+        // column-map increments
+        NavigableMap<byte[], Long> increments = new TreeMap<>(Bytes.BYTES_COMPARATOR);
+
+        for (Map.Entry<byte[], byte[]> entry : columns.entrySet()) {
+          // column is timestamp, do a get on the new table
+          byte[] value = v3MetricsTable.get(rowKey, entry.getKey());
+          if (value == null) {
+            LOG.debug("Trying checkAndPut");
+            if (v3MetricsTable.checkAndPut(rowKey, entry.getKey(), new byte[0], entry.getValue())) {
+              LOG.debug("checkAndPut Successful");
+              gaugeDeletes.add(entry.getKey());
+            } else {
+              LOG.debug("checkAndPut failed");
```

Comment:
What is the idea here? If the checkAndPut fails, the column remains unmodified and will be migrated in the next iteration? It would be good to explain the approach in a comment